File: js/views/sessions.js
import { getSessions, getPlayerById } from '../store.js';
import { getTrackById, getCarById } from '../data.js';

export function render() {
  const sessions = getSessions();

  if (sessions.length === 0) {
    return `
      <div class="sessions-view">
        <div class="view-header">
          <h1 class="section-title">Carreras</h1>
          <a href="#/sessions/new" class="btn btn-primary">Nueva Carrera</a>
        </div>
        <div class="empty-state small">
          <p>No hay carreras registradas.</p>
        </div>
      </div>
    `;
  }

  return `
    <div class="sessions-view">
      <div class="view-header">
        <h1 class="section-title">Carreras</h1>
        <a href="#/sessions/new" class="btn btn-primary">Nueva Carrera</a>
      </div>
      <div class="race-list">
        ${sessions.map(s => {
          const track = getTrackById(s.trackId);
          const car = getCarById(s.carId);
          const winner = s.results[0];
          const winnerPlayer = winner ? getPlayerById(winner.playerId) : null;
          return `
            <a href="#/sessions/${s.id}" class="race-card">
              <div class="race-info">
                <span class="race-date">${new Date(s.date).toLocaleDateString('es-MX')}</span>
                <span class="race-track">${track ? track.name : s.trackId}</span>
                <span class="race-car">${car ? car.name : s.carId}</span>
              </div>
              <div class="race-meta">
                <span class="race-players">${s.results.length} pilotos</span>
                <div class="race-winner">
                  <span class="winner-label">P1</span>
                  <span class="winner-name">${winnerPlayer ? winnerPlayer.name : 'Desconocido'}</span>
                </div>
              </div>
            </a>
          `;
        }).join('')}
      </div>
    </div>
  `;
}

export function init() {}

File: volantazo/lib/screens/sessions.dart
import 'package:flutter/material.dart';
import '../services/storage_service.dart';
import '../data/tracks.dart';
import '../data/cars.dart';
import '../theme.dart';

class SessionsScreen extends StatefulWidget {
  final StorageService storage;

  const SessionsScreen({super.key, required this.storage});

  @override
  State<SessionsScreen> createState() => _SessionsScreenState();
}

class _SessionsScreenState extends State<SessionsScreen> {
  @override
  Widget build(BuildContext context) {
    final sessions = widget.storage.getSessions();

    if (sessions.isEmpty) {
      return const Center(
        child: Text('No hay carreras registradas.', style: TextStyle(color: AppTheme.textSecondary)),
      );
    }

    return ListView.builder(
      padding: const EdgeInsets.all(16),
      itemCount: sessions.length,
      itemBuilder: (context, i) {
        final s = sessions[i];
        final track = getTrackById(s.trackId);
        final car = getCarById(s.carId);
        final winner = s.results.isNotEmpty ? s.results.first : null;
        final winnerName = winner != null && !winner.dnf
            ? (widget.storage.getPlayerById(winner.playerId)?.name ?? '?')
            : null;

        return Card(
          margin: const EdgeInsets.only(bottom: 8),
          child: InkWell(
            borderRadius: BorderRadius.circular(8),
            onTap: () => Navigator.pushNamed(context, '/session', arguments: s.id).then((_) => setState(() {})),
            child: Padding(
              padding: const EdgeInsets.all(16),
              child: Row(
                children: [
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(track?.name ?? s.trackId, style: const TextStyle(fontWeight: FontWeight.w700)),
                        const SizedBox(height: 4),
                        Text(
                          '${car?.name ?? s.carId} · ${s.date} · ${s.results.length} pilotos',
                          style: const TextStyle(color: AppTheme.textSecondary, fontSize: 13),
                        ),
                      ],
                    ),
                  ),
                  if (winnerName != null) ...[
                    Container(
                      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 2),
                      decoration: BoxDecoration(color: AppTheme.gold, borderRadius: BorderRadius.circular(4)),
                      child: const Text('P1', style: TextStyle(color: Colors.black, fontWeight: FontWeight.w700, fontSize: 12)),
                    ),
                    const SizedBox(width: 8),
                    Text(winnerName, style: const TextStyle(fontWeight: FontWeight.w600)),
                  ],
                ],
              ),
            ),
          ),
        );
      },
    );
  }
}
